File: lib/data/model/news/news.dart
import 'package:json_annotation/json_annotation.dart';
part 'news.g.dart';

@JsonSerializable()
class News {
  final String newsId;
  final String? headline;
  final String title;
  final String description;
  final String? imageUrl;
  final String category;
  final int likeCount;
  final bool isActive;
  final DateTime publishedAt;

  News({
    required this.newsId,
    required this.headline,
    required this.title,
    required this.description,
    required this.imageUrl,
    required this.category,
    required this.likeCount,
    required this.isActive,
    required this.publishedAt,
  });

  factory News.fromJson(Map<String, dynamic> json) => _$NewsFromJson(json);
  Map<String, dynamic> toJson() => _$NewsToJson(this);
}

File: Models/News.cs
namespace AhaanNews.Models
{
    public class News
    {
        public Guid NewsId { get; set; } = Guid.NewGuid();
        public string? Headline { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string? ImageUrl { get; set; }
        public string Category { get; set; }
        public int LikeCount { get; set; } = 0;
        public bool IsActive { get; set; } = true;
        public DateTime PublishedAt { get; set; }

        public ICollection<UserLikedNews> LikedBy { get; set; }
        public ICollection<UserBookmarkedNews> BookmarkedBy { get; set; }
        public ICollection<NewsVideo> Videos { get; set; }
    }
}
